editor/src/clj/editor/app_view.clj

```diff
@@ -2011,26 +2011,47 @@ If you do not specifically require different script states, consider changing th
       (ui/on-closed! stage (fn [_] (dispose-scene-views! app-view)))
       app-view)))
 
+(defn- make-info-box! []
+  (let [info-panel (HBox.)
+        left-label (Label. "This file is part of a library and cannot be saved.  ")
+        right-link (Hyperlink. "Read more…")
+        spacer (Region.)]
+    (HBox/setHgrow spacer Priority/ALWAYS)
+    (.getStyleClass info-panel)
+    (ui/set-style! info-panel "info-panel" true)
+    (ui/set-style! left-label "info-panel-label" true)
+    (ui/set-style! right-link "info-panel-link" true)
+    (.setOnAction right-link (ui/event-handler _ (ui/open-url "https://www.defold.com/manuals/editor/")))
+    (.addAll  (.getChildren info-panel) (Arrays/asList (into-array Node [left-label spacer right-link])))
+    info-panel))
+
 (defn- make-tab! [app-view prefs workspace project resource resource-node
                   resource-type view-type make-view-fn ^ObservableList tabs
                   open-resource opts]
-  (let [parent     (AnchorPane.)
-        tab        (doto (Tab. (tab-title resource false))
-                     (.setContent parent)
-                     (.setTooltip (Tooltip. (or (resource/proj-path resource) "unknown")))
-                     (ui/user-data! ::view-type view-type))
+  (let [parent (AnchorPane.)
+        res-read-only? (if (resource/read-only? resource) true false)
+        tab-content (if res-read-only? (VBox.) parent)
```

Comment:
I mean... all the changes in this function could be simplified to something like
```clj
tab-content (if (resource/read-only? resource)
              (doto (VBox.)
                (ui/children! [(make-info-box!)
                               (doto parent (VBox/setVgrow Priority/ALWAYS))]))
              parent)
```

Reply:
smart!